### DISCUSSION TITLE
Sound gets disabled in Sudo and VC4

### DISCUSSION THREAD

**Discussion by ScytheandGenesisGames** [BlitterStudio/amiberry#1015] [General]
There's this KMS Graphics driver that uses the full 60hz for Vsync and the only problem with it using with sudo (for the GPIO Pins) sound is disabled when I try to use the VC4 HDMI Audio driver.

I don't know why it's such a problem with the KMS Driver and the Sound Driver it uses.

I also tried enabling the sound and it still "Disables" the sound in Sudo running the program.

**Comment by solskogen** [comment]
Install pulseaudio. I guess you're running lite, so that is not installed. pulseaudio is required when using the kms-driver.

**Comment by ScytheandGenesisGames** [comment]
> Install pulseaudio. I guess you're running lite, so that is not installed. pulseaudio is required when using the kms-driver.

not running the lite version it's the full fledged OS with everything, I don't know if using KMS is better or not to fix the syncing issue.

**Comment by solskogen** [comment]
Does `speaker-test -c2` produce any sound?

**Comment by ScytheandGenesisGames** [comment]
> Does `speaker-test -c2` produce any sound?

that works when set to hdmi on main it's a bug when starting Pulseaudio as the driver is installed already - it "only" applies to the headphones jack which I don't wish to use. I use a launcher to open Amiberry in Sudo mode I don't know a command that forces the Hdmi's pulseaudio to work if I use the KMS driver.

**Comment by solskogen** [comment]
It's a bit out of scope for amiberry. Try the Raspberry Pi forums.

**Comment by ScytheandGenesisGames** [comment]
I'll try what raspberry pi calls the FAKE KMS which enables OpenGL but doesn't use VC4 Sound but instead PulseAudio

**Comment by solskogen** [comment]
I've got no problem getting sound from HDMI when using KMS and pulseaudio on both Pi4 and Pi400, but I don't know what you've done to your system so it's a bit hard for me to help.

**Reply by ScytheandGenesisGames** [reply]
I never used Full KMS at all I prefered using the non-GL so I input the overlay to use the Fake KMS driver just for Pulseaudio to work it has to be the Raspberry pi's terminal Config that changes things for you must be bugged.

I've just added the Fake KMS in my config file for OpenGL with PulseAudio as the Full KMS sets the sound driver to VC4 no matter if PulseAudio is installed.

**Reply by solskogen** [reply]
Could you please rephrase? I don't understand.

**Reply by ScytheandGenesisGames** [reply]
I just used the Fake KMS OpenGL Driver in the Pi's config file so it uses Pulseaudio instead of the Full KMS VC4 Sound driver

dtoverlay=vc4-fkms-v3d - this enables the OpenGL and not the VC4 audio driver it wanted to use so I stick to that for the full Vsync Amiberry uses.
